services/wallet/api.go

```diff
@@ -896,10 +896,19 @@ func (api *API) getVerifiedWalletAccount(address, password string) (*account.Sel
 	}, nil
 }
 
+type InjectedMessenger interface {
+	NewWalletConnectV2SessionCreatedNotification(session walletconnect.Session) error
+}
+
```

Comment:
Well this is a very specific interface! 😄 

1. Interface name shouldn't describe how it's used, but the word "Injected" explicitly means some specific usage.
Instead, it should describe the what's the interface about, in this case it's about adding some activity center notifications.

2. I don't like the idea of an interface for one specific need.
@dlipicar maybe you know if we already have some wallet-messenger communication?

Reply:
well i'm not an expertise in this area, but Rashawn mentioned in description that he did to avoid import cycle, but if you have any suggestion please let me know so i will apply.